src/sql/workbench/api/node/mainThreadConnectionManagement.ts

```diff
@@ -110,6 +123,30 @@ export class MainThreadConnectionManagement implements MainThreadConnectionManag
 		return connection;
 	}
 
+	private convertToConnectionProfile(profile: IConnectionProfile): azdata.connection.ConnectionProfile {
+		if (!profile) {
+			return undefined;
+		}
+
+		profile = this._connectionManagementService.removeConnectionProfileCredentials(profile);
+		let connection: azdata.connection.ConnectionProfile = {
```

Comment:
Possibly irrelavent depending on answer to above comment, but instead of doing this you could also just not set the password below and save a whole lot of round tripping